### PR TITLE
Use full path for verified files

### LINKED ISSUE #4: Accepting received does not work if using `VerifierSettings.DerivePathInfo`
Trying to accept the received results as verified seems to do nothing.

Having seen the changes you made in 2021.2.1 i noticed that this will never work in our case, or in case someone uses a different folder to store the files.
We are using `VerifierSettings.DerivePathInfo` to change the path to not be side-by-side with the test file, but produce a separate directory beside the test file. But the exception/error text produced by verify only contains the filename, not the path.
As the plugin uses that text to extract the filename, it doesn't currently seem to be possible to solve this only on the plugin side.

@SimonCropp Maybe you have an idea on how to solve this? Maybe an option to use the full paths inside `ProcessNotEquals` instead of `Path.GetFileName`? Or maybe append the full paths at the bottom of the result?

Versions used:
Rider: 2021.2.1
Plugin: 2021.2.1

### LINKED ISSUE #5: Rider "Verify Support" plugin buttons does nothing
This is an almost straight copy from here: https://github.com/VerifyTests/Verify/issues/452

#### Describe the bug
I'm using Verify to test a source generator. The test itself runs fine and has the expected output. However, trying to use the "Accept received" and "Compare Accepted/Receives" buttons does nothing. As in nothing happens when I click those buttons. I have ensured that tests are allowed to start sub-processes. When the test fails a diff tool does start, so Rider can clearly be found by Verify. 

Logs from Rider, where I only started the IDE, found a failing test and clicked those buttons. 
[rider verify logs.zip](https://github.com/VerifyTests/Verify/files/7768264/rider.verify.logs.zip)

And video in case my rambling didn't make any sense:

https://user-images.githubusercontent.com/1499810/147223348-8945bd14-0f98-42ea-989e-4f57801441d4.mp4

#### Minimal Repro

Here is a somewhat minimal solution that has a source generator and everything. The buttons still don't work. 
[VerifyBug.zip](https://github.com/VerifyTests/Verify/files/7768321/VerifyBug.zip)

Rider version:
```
JetBrains Rider 2021.3.1
Build #RD-213.5744.299, built on December 8, 2021
Licensed to Rasmus Hjelmberg Hansen
Subscription is active until September 19, 2022.
Runtime version: 11.0.13+7-b1751.19 amd64
VM: Dynamic Code Evolution 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
.NET Framework 4.0.30319.42000
GC: G1 Young Generation, G1 Old Generation
Memory: 8192M
Cores: 12
Registry:
    ide.mac.bigsur.window.with.tabs.enabled=false
    rdclient.asyncActions=false
    indexing.enable.entity.provider.based.indexing=false
    ide.new.project.model.index.case.sensitivity=true

Non-Bundled Plugins:
    verify-rider (2021.3.0)
    org.antlr.intellij.plugin (1.17)
    idea.plugin.protoeditor (213.5744.224)
    com.intellij.resharper.StructuredLogging (2021.3.0.177)
    com.intellij.resharper.HeapAllocationsViewer (2021.3.0)
    com.intellij.ideolog (203.0.27.0)
    com.intellij.plugin.adernov.powershell (2.0.9)

```

Verify Support plugin version: `2021.3.0`

### PR DESCRIPTION
**Pull request by SimonCropp**
fixes #5 fixes #4

### CODE REVIEW

File: src/dotnet/Directory.Build.props [hunk 1]
```diff
@@ -5,10 +5,13 @@
...
     <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
     <ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>None</ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>
-
     <BaseIntermediateOutputPath>obj\$(MSBuildProjectName)\</BaseIntermediateOutputPath>
     <DefaultItemExcludes>$(DefaultItemExcludes);obj\**</DefaultItemExcludes>
     <OutputPath>bin\$(MSBuildProjectName)\$(Configuration)\</OutputPath>
+    <LangVersion>10.0</LangVersion>
+    <WaveVersionBase>$(SdkVersion.Substring(2,2))$(SdkVersion.Substring(5,1))</WaveVersionBase>
+    <WaveVersion>$(WaveVersionBase).0.0$(SdkVersion.Substring(8))</WaveVersion>
+    <UpperWaveVersion>$(WaveVersionBase).9999.0</UpperWaveVersion>
```
Comment (matkoch) [inline src/dotnet/Directory.Build.props:14]: This I would move back to its own group. Mostly to keep sync to the template.

File: src/dotnet/ReSharperPlugin.Verify/VerifyAcceptAction.cs [hunk 2]
```diff
@@ -69,13 +69,17 @@ public void Execute(IDataContext context, DelegateExecute nextExecute)
...
-                var verifiedFile = (projectFile.Location.Directory / verifiedFileName.NotNull("verifiedFileName")).FullPath;
+                var exceptionLines = result.GetExceptionChunk(2)
+                    .SplitByNewLine()
+                    //Received path and Verified path are at the end of the exception message
+                    .Reverse()
+                    .ToArray();
+                var (receivedFile, verifiedFile) = (
+                    exceptionLines.FirstOrDefault(x => x.StartsWith("Received path: "))?.TrimFromStart("Received path: "),
+                    exceptionLines.FirstOrDefault(x => x.StartsWith("Verified path: "))?.TrimFromStart("Verified path: "));
+
```
Comment (matkoch) [inline src/dotnet/ReSharperPlugin.Verify/VerifyAcceptAction.cs:44]: This means older versions aren't supported, right?

File: src/dotnet/Plugin.props [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project>
 
   <PropertyGroup>
-    <SdkVersion>2021.3.1</SdkVersion>
+    <SdkVersion>2021.4.0</SdkVersion>
```
Comment (matkoch) [inline src/dotnet/Plugin.props:4]: While not ideal, major and minor are dictated by the R#/Rider version – at least for the plugins that I maintain. This is to reflect what version it belongs to.

Comment (matkoch) [inline src/dotnet/Plugin.props:4]: What SDK version is this supposed to be? Currently `2021.3.2` is highest.

Comment (SimonCropp) [reply src/dotnet/Plugin.props:4]: oops. i thought that was the version number

File: src/dotnet/Directory.Build.props (hunk 1 above)
Comment (SimonCropp) [reply src/dotnet/Directory.Build.props:14]: ok

File: src/dotnet/ReSharperPlugin.Verify/VerifyAcceptAction.cs (hunk 2 above)
Comment (SimonCropp) [reply src/dotnet/ReSharperPlugin.Verify/VerifyAcceptAction.cs:44]: it does

**Comment by SimonCropp** [comment]
good to go?

**Comment by matkoch** [comment]
If I understand correctly, then the new implementation has to parse the full exception output to gather received/accepted file names. This should work, but it does add some unpredictable performance hit.

**Comment by SimonCropp** [comment]
> This should work, but it does add some unpredictable performance hit.

OK. i wasnt aware it was expensive to parse the exception message. how about i move the to be near the top. ie full path directly below each partial path?

**Comment by matkoch** [comment]
If it was guaranteed to be always at the very top, that would be awesome :)

**Comment by SimonCropp** [comment]
can we access the exception data property? then we could access the paths with a key lookup

**Comment by SimonCropp** [comment]
@matkoch bump on my last comment

**Comment by SimonCropp** [comment]
or can we access custom properties?

**Comment by matkoch** [comment]
Seems that only `exception.ToString()` is available :/

**Comment by SimonCropp** [comment]
how do i explicitly call `ToString`?

**Comment by SimonCropp** [comment]
is there any docs for UnitTestResultData.GetExceptionChunk ?

**Comment by SimonCropp** [comment]
i think we need an alternative to using the exception message. the exception message is displayed to the user, so it needs to be human readable and make sense. with the amount of information we need to pass to the addin, it pollutes the message displayed to the user. and not that we have not handled a few edge cases yet, eg tests that produce multiple snapshot files and deletion of redundant snapshots. 

can the sdk be extended to have a feature where we can use Exception.Data or a custom property to pass a key/value list from the test to the addin?

**Comment by matkoch** [comment]
I forwarded your slack message. Can't really comment on that.

My thinking is that anything that would go into the `Data` property could as well be printed via `exception.ToString()`.

**Comment by SimonCropp** [comment]
>  wouldn't you want to show this as part of the message to let the user know?

yes but showing that in a way that is human readable and machine parsable is difficult.

**Comment by SimonCropp** [comment]
raised a feature request in the plugin api https://youtrack.jetbrains.com/issue/RSPL-6990

**Comment by matkoch** [comment]
lets wait for that then?

**Comment by SimonCropp** [comment]
@matkoch eta? days/weeks/months?

**Comment by SimonCropp** [comment]
@matkoch and can i help? eg sign an nda and implement it myself?

**Comment by matkoch** [comment]
youtrack issue would be a better place to ask, but i'm afraid the real bottleneck is the release cadence.
